hoomd/mpcd/CollisionMethod.cc

```diff
@@ -102,11 +119,17 @@ void mpcd::CollisionMethod::collide(uint64_t timestep)
         if (m_exec_conf->isCUDAEnabled())
             {
             storeInitialEmbeddedGroupVelocitiesGPU(timestep);
+            // thermalize constituent particles to add energy to the rigid body
+            drawVelocitiesConstituentParticlesGPU(timestep);
+            getNetVelocityRigidBodyGPU(timestep);
+            applyThermalizedVelocityVectorsGPU(timestep);
```

Comment:
These should be one `thermalizeConstituentParticlesGPU` method. You can use `{}` to scope array handles inside that function if needed. Same comment about the order of operations though.